openshift/templates/cakephp-mysql.json

```diff
@@ -354,6 +354,12 @@
   ],
   "parameters": [
     {
+      "name": "NAME",
+      "displayName": "Name",
+      "description": "The name assigned to all of the objects defined in this template.",
```

Comment:
Actually i think i'm wrong.  As you note, the DB object names are already parameterized.

The items you listed don't contribute to the name of a top-level object, so they do not need to be parameterized for uniqueness.

you also didn't need to parameterize the container names in the frontend objects (or any references to the container names, ie via the triggers)